action.yml

```diff
@@ -1,121 +1,222 @@
     required: false
   report_code:
-    description: 'The code of the report. If unsure, do not include'
+    description: 'The code of the report if using local upload. If unsure, leave default. Read more here https://docs.codecov.com/docs/the-codecov-cli#how-to-use-local-upload'
+    required: false
+  report_type:
+    description: 'The type of file to upload, coverage by default. Possible values are "testing", "coverage".'
     required: false
-  root_dir:
-    description: 'Used when not in git/hg project to identify project root directory'
+  root_folder:
+    description: 'Root folder from which to consider paths on the network section. Defaults to current working directory.'
     required: false
+  skip_validation:
+    description: 'Skip integrity checking of the CLI. This is NOT recommended.'
+    required: false
+    default: 'false'
   slug:
-    description: 'Specify the slug manually (Enterprise use)'
+    description: 'Set to the owner/repo slug used instead of the private repo token. Only applicable to some Enterprise users.'
+    required: false
+  swift_project:
+    description: 'Specify the swift project name. Useful for optimization.'
+    required: false
+  token:
+    description: 'Repository Codecov token. Used to authorize report uploads'
     required: false
   url:
-    description: 'Specify the base url to upload (Enterprise use)'
+    description: 'Set to the Codecov instance URl. Only applicable to some Enterprise users.'
     required: false
   use_legacy_upload_endpoint:
-    description: 'Use the legacy upload endpoint'
+    description: 'Use the legacy upload endpoint.'
     required: false
+    default: 'false'
   use_oidc:
     description: 'Use OIDC instead of token. This will ignore any token supplied'
     required: false
+    default: 'false'
   verbose:
-    description: 'Specify whether the Codecov output should be verbose'
+    description: 'Use verbose logging in the CLI'
     required: false
+    default: 'false'
   version:
-    description: 'Specify which version of the Codecov CLI should be used. Defaults to `latest`'
+    description: "Which version of the Codecov CLI to use (defaults to 'latest')"
     required: false
+    default: 'latest'
   working-directory:
     description: 'Directory in which to execute codecov.sh'
     required: false
+
+steps:
+  - run:
+      name: Upload to Codecov
+      command: <<include(scripts/dist/codecov.sh)>>
+      when: << parameters.when >>
 branding:
   color: 'red'
   icon: 'umbrella'
 runs:
-  using: 'node20'
-  main: 'dist/index.js'
+  using: "composite"
+  steps:
+    - name: Set safe directory
+      if: ${{ inputs.set_safe_directory != 'true' }}
+      shell: bash
+      run: |
+        git config --global --add safe.directory ${{ github.workspace }}
+
+    - name: Get and set token
+      shell: bash
+      run: |
+        if [ ${{ inputs.use_oidc }} == 'true' ];
+        then
+          # {"count":1984,"value":"***"}
```

Comment:
What is this?

Reply:
A comment to future Tom to know what the format of the `response` is